### ISSUE TITLE
Fully general `_field` predicate support in InfluxRPC

### ISSUE THREAD

**Issue by alamb** [influxdata/influxdb_iox#5310]
__Use case:__
It is valid (though I don't know if it is at all common) to send a predicate like

```sql
(_field = 'foo') OR (bar < 5.0)
```

The meaning of this is that all series with fields named `foo` or any series that have a value of the `bar` field that is less than 5. 

IOx's current handing of the `_field` predicates does not handle this case, it only handles the case where the `_field` predicate can be evaluated at plan time (without access to any of the actual data). 

It can handle fairly sophisticated predicates like
```sql
(_field = 'foo' OR _field = 'bar') AND bar < 5.0
```
where the difference is that we can evaluate `(_field = 'foo' OR _field = 'bar')` at planning time (see https://github.com/influxdata/influxdb_iox/pull/5301 for how this was accomplished)

__Proposal:__
The comments here sketch out some way to do it https://github.com/influxdata/influxdb_iox/issues/5177#issuecomment-1192924105

TLDR is I think to handle the general case, we will have to run a specialized plan for *each* field column with predicate rewritten with all _field references replaced with the actual field name and then merge the output of all those plans together.

__Current behaviour:__
Get an error

__Desired behaviour:__
Plans to run

__Alternatives considered:__
Leave it unimplemented (it is not clear if these types of predicates are ever actually generated)

**Comment by alamb** [comment]
See the test_read_filter_unsupported_field_predicate test for an example

**Comment by alamb** [comment]
https://github.com/influxdata/conductor/issues/1112 has an example of a customer query in production that shows this pattern

**Comment by Marwes** [comment]
I think this may cover https://github.com/influxdata/influxdb_iox/issues/5428 .

This test failure from running `get_api_usage`  might also be related (this at least returns an error, instead of returning the wrong data).

```
	vendor/github.com/influxdata/flux/stdlib/testing/usage/api_test.flux: get_api_usage ... fail: query error: rpc error: code = InvalidArgument desc = Error creating group plans for database '54425547585a6420_fc40686c0f64900f': Error during planning: read_group error: gRPC planner got error creating predicates: Error during planning: Unsupported _field predicate: CASE WHEN #endpoint IS NULL THEN Utf8("") ELSE #endpoint END = Utf8("/api/v2/write") AND #_field = Utf8("req_bytes") AND RegexNotMatch(CASE WHEN #hostname IS NULL THEN Utf8("") ELSE #hostname END, Utf8("^gateway-internal")) OR CASE WHEN #endpoint IS NULL THEN Utf8("") ELSE #endpoint END = Utf8("/api/v2/query") AND #_field = Utf8("resp_bytes")
```
https://github.com/influxdata/flux/blob/51b7e698d34fb95070b78f4841929d1d2b1d8f21/stdlib/testing/usage/api_test.flux#L1687-L1709

Maybe `filter_drop_empty` as well

```
vendor/github.com/influxdata/flux/stdlib/universe/filter_drop_empty_test.flux: filter_drop_empty ... fail: Table: keys: [_start, _stop, _field, _measurement, host, name]
                   _start:time                      _stop:time           _field:string     _measurement:string             host:string             name:string            _diff:string                  _value:int  
------------------------------  ------------------------------  ----------------------  ----------------------  ----------------------  ----------------------  ----------------------  --------------------------  
                                                                               io_time                  diskio              host.local                   disk2                       +                           0  
```
https://github.com/influxdata/flux/blob/51b7e698d34fb95070b78f4841929d1d2b1d8f21/stdlib/universe/filter_drop_empty_test.flux#L37-L47

**Comment by alamb** [comment]
hi @Marwes 

I agree that the errors look related. Specifically, if you see this

> gRPC planner got error creating predicates: Error during planning: Unsupported _field predicate: 

That is a telltale sign of this bug (I have updated the PR description to also have this error message)

**Comment by alamb** [comment]
@crepererum  mentioned this as something that was annoying on IOx producton

**Comment by jacobmarble** [comment]
From the Flux team:
> this issue is now blocking our efforts to make meta queries in Flux work with IOx

**Comment by alamb** [comment]
Update here is that @jacobmarble  plans to discuss various alternatives with stakeholders -- I will hold off on implementing this until then

**Comment by jacobmarble** [comment]
Plan: deprioritize until some customer needs this behavior corrected
